### ISSUE TITLE
tests/test_mutable_multidict.py::TestCIMutableMultiDict::test_add[multidict._multidict] segfaults on py3.12

### ISSUE THREAD

**Issue by mgorny** [aio-libs/multidict#868]
## Long story short

When running the test suite on Python 3.12.0b4, I get a segfault.

## Expected behaviour

Passing tests ;-).

## Actual behaviour

```python-traceback
tests/test_mutable_multidict.py::TestCIMutableMultiDict::test_add[multidict._multidict] Fatal Python error: Segmentation fault

Current thread 0x00007f721f940740 (most recent call first):
  File "/tmp/multidict/tests/test_mutable_multidict.py", line 333 in test_add
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/python.py", line 194 in pytest_pyfunc_call
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 80 in _multicall
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 433 in __call__
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/python.py", line 1788 in runtest
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 169 in pytest_runtest_call
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 80 in _multicall
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 433 in __call__
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 262 in <lambda>
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 341 in from_call
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 261 in call_runtest_hook
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 222 in call_and_report
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 133 in runtestprotocol
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 114 in pytest_runtest_protocol
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 80 in _multicall
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 433 in __call__
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/main.py", line 349 in pytest_runtestloop
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 80 in _multicall
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 433 in __call__
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/main.py", line 324 in _main
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/main.py", line 270 in wrap_session
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/main.py", line 317 in pytest_cmdline_main
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 80 in _multicall
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 112 in _hookexec
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 433 in __call__
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py", line 166 in main
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py", line 189 in console_main
  File "/tmp/multidict/.venv/lib/python3.12/site-packages/pytest/__main__.py", line 5 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main

Extension modules: multidict._multidict (total: 1)
Segmentation fault (core dumped)
```

## Steps to reproduce

```
python3.12 -m venv .venv
. .venv/bin/activate
pip install -e . pytest-cov
python -m pytest -s
```

## Your environment

Gentoo Linux amd64
Python 3.12.0b4
Reproduced on a05f1f8bf22f9fe9d480da88128565bf3ac476f3 and v6.0.4.

```console
$ pip list
Package    Version Editable project location
---------- ------- -------------------------
coverage   7.2.7
iniconfig  2.0.0
multidict  6.0.4   /tmp/multidict
packaging  23.1
pip        23.2.1
pluggy     1.2.0
pytest     7.4.0
pytest-cov 4.1.0
```

**Comment by webknjaz** [comment]
@mgorny thanks for the report! Can you still reproduce it? The CI is running against Python 3.12 just fine.

**Comment by mgorny** [comment]
Yes, FWICS it was fixed by 780a1e1deed31bd0d7b01db7dee931ec1365006f.

**Comment by webknjaz** [comment]
Great, thanks!
